src/lib/support/StringBuilder.h

```diff
@@ -89,13 +90,23 @@ class StringBuilderBase
     }
 };
 
-/// a preallocated sized string builder
-template <size_t kSize>
+/// A preallocated sized string builder
+/// According to the specification there are fields with size up to 256
```

Comment:
This needs some updates: string builder has no direct reference in the specification, so we should probably say that a very common use case is to wrap around field content for logging purposes.

This does feel awkward though ... I wish we had a separate method instead like:

```cpp
StringBuilder<257> NullTerminated(CharSpan rawData) { // ...
```

that seems to be more obvious and would not tie in a StringBuilder with the chip specification directly.